server/__init__.py

```diff
@@ -274,6 +276,26 @@ def validateFileLink(event):
     event.preventDefault().addResponse(doc)
 
 
+def updateNotification(event):
+    """
+    Update the Whole Tale task notification for a job, if present.
+    """
+
+    job = event.info['job']
+    if job['progress'] and 'wt_notification_id' in job:
+        state = JobStatus.toNotificationStatus(job['status'])
+        notification = Notification().load(job['wt_notification_id'])
+
+        if notification['data']['message'] != job['progress']['message']:
```

Comment:
What happens if job status changes (e.g. it fails)? Does it affect progress message, i.e. will it change so that notification status is updated accordingly?